opm/parser/eclipse/EclipseState/Tables/SimpleTable.hpp

```diff
@@ -60,11 +60,15 @@ namespace Opm {
          */
         double evaluate(const std::string& columnName, double xPos) const;
 
+        /// throws std::invalid_argument if jf != m_jfunc
+        void assertJFuncPressure(const bool jf) const;
```

Comment:
you think it's better to have this function in five different classes than in one superclass?  What about the `m_jfunc` boolean flag?  Should that go to the five classes as well?